firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -278,16 +278,22 @@ String getName() {
     return Tasks.call(backgroundExecutor, this::deleteFirebaseInstallationId);
   }
 
-  /** Register a callback {@link FidListener} to receive fid changes. */
+  /**
+   * Register a callback {@link FidListener} to receive fid changes.
+   *
+   * @hide
+   */
   @NonNull
   @Override
-  public synchronized HandleFidListener registerFidListener(@NonNull FidListener listener) {
-    this.fidListeners.add(listener);
-    return new HandleFidListener() {
+  public synchronized FidListenerHandle registerFidListener(@NonNull FidListener listener) {
+    fidListeners.add(listener);
+    return new FidListenerHandle() {
       @Override
-      public void unregister(FidListener fidListener) {
-        if (fidListeners.contains(fidListener)) {
-          fidListeners.remove(fidListener);
+      public void unregister() {
+        synchronized (FirebaseInstallations.this) {
+          if (fidListeners.contains(listener)) {
```

Comment:
nit: consider removing the `contains` check as remove handles it internally.

Reply:
Thanks. Done